upgrade/scripts/upgrade/util/upgrade-spire.sh

```diff
@@ -22,25 +22,82 @@
-            yq d -i $BUILDDIR/spireupgrade.yaml "spec.charts.(name==$chart)"
-        fi
-    done
+function sshnh() {
+  /usr/bin/ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null "$@"
+}
 
-    yq w -i $BUILDDIR/spireupgrade.yaml "metadata.name" "spireupgrade"
-    yq d -i $BUILDDIR/spireupgrade.yaml "spec.sources"
+for node in $(kubectl get nodes -o name | cut -d"/" -f2) $(ceph node ls | jq -r '.[] | keys[]' | sort -u); do
+  echo "$node: Stopping spire-agent"
+  sshnh "$node" systemctl stop spire-agent
+  sshnh "$node" rm "${svidkey}" "${bundleder}" "${agentsvidder}" "${jointoken}" || true
 
-    loftsman ship --charts-path "${CSM_ARTI_DIR}/helm/" --manifest-path $BUILDDIR/spireupgrade.yaml
-}
+  sshnh "$node" zypper install -y spire-agent
+
+  echo "$node: Starting spire-agent"
+  sshnh "$node" systemctl start spire-agent
+done
+
+echo "Uninstalling spire helm chart"
+helm uninstall -n spire spire
+echo "Uninstalling spire PVCs"
+kubectl delete -n spire pvc spire-data-spire-server-0 spire-data-spire-server-1 spire-data-spire-server-2
+
+RETRY=0
+MAX_RETRIES=30
+RETRY_SECONDS=5
+
+until [ "$(kubectl get pods -n spire --no-headers | wc -l)" -ne 0 ]; do
+  if [[ "$RETRY" -lt "$MAX_RETRIES" ]]; then
+    RETRY="$((RETRY + 1))"
+    echo "Spire is still in the process of being uninstalled. Will retry after $RETRY_SECONDS seconds. ($RETRY/$MAX_RETRIES)"
+  else
+    echo "Spire did not completely uninstall after $(echo "$RETRY_SECONDS" \* "$MAX_RETRIES" | bc) seconds."
+    exit 1
+  fi
+  sleep "$RETRY_SECONDS"
+done
+
+echo "Installing new spire helm chart"
+BUILDDIR="/tmp/build"
+mkdir -p "$BUILDDIR"
+kubectl get secrets -n loftsman site-init -o jsonpath='{.data.customizations\.yaml}' | base64 -d >"${BUILDDIR}/customizations.yaml"
+manifestgen -i "${CSM_ARTI_DIR}/manifests/sysmgmt.yaml" -c "${BUILDDIR}/customizations.yaml" -o "${BUILDDIR}/spireupgrade.yaml"
+charts="$(yq r $BUILDDIR/spireupgrade.yaml 'spec.charts[*].name')"
+for chart in $charts; do
+  if [[ $chart != "spire" ]]; then
+    yq d -i $BUILDDIR/spireupgrade.yaml "spec.charts.(name==$chart)"
+  fi
+done
+
+yq w -i $BUILDDIR/spireupgrade.yaml "metadata.name" "spireupgrade"
+yq d -i $BUILDDIR/spireupgrade.yaml "spec.sources"
+
+loftsman ship --charts-path "${CSM_ARTI_DIR}/helm/" --manifest-path $BUILDDIR/spireupgrade.yaml
+
+RETRY=0
+MAX_RETRIES=30
+RETRY_SECONDS=10
+
+until kubectl exec -itn spire spire-server-0 --container spire-server -- ./bin/spire-server healthcheck | grep -q 'Server is healthy'; do
+  if [[ "$RETRY" -lt "$MAX_RETRIES" ]]; then
+    RETRY="$((RETRY + 1))"
+    echo "spire-server is not ready. Will retry after $RETRY_SECONDS seconds. ($RETRY/$MAX_RETRIES)"
+  else
+    echo "spire-server did not start after $(echo "$RETRY_SECONDS" \* "$MAX_RETRIES" | bc) seconds."
+    exit 1
+  fi
+  sleep "$RETRY_SECONDS"
+done
 
-deploySpire
+echo "Joining storage nodes to spire"
+/opt/cray/platform-utils/spire/fix-spire-on-storage.sh
```

Comment:
Should the exit code of this be checked? I know that if it exits non-0, this overall script will exit non-0, but perhaps we'd also want to print an error message explaining what failed?